src/Random.h

```diff
@@ -0,0 +1,35 @@
+/*
+ * Random.h
+ *
+ *  Created on: 6 Aug 2020
+ *      Author: gc1610
+ */
+
+#ifndef SRC_RANDOM_H_
+#define SRC_RANDOM_H_
+
+#include <vector>
+
+class RandomInterface {
+public:
+    virtual std::vector<size_t> bernoulli(size_t, double) = 0;
+    virtual ~RandomInterface() = default;
```

Comment:
Yes, it's a side-effect.

We need virtual methods so that the mocking framework can override them.

By default, C++ compilers will implicitly decare non-virtual destructors. But base classes should always have virtual destructors. Because if we (or the R garbage collector) ever wants to delete a `Random` object through a `RandomInterface*` it's undefined.

So we explicitly say that the destructor is virtual.

This shouldn't be an issue at the moment since `Random` is a static singleton and so won't be `delete`d. But it's good practice anyway.